uaclient/tests/test_cli.py

```diff
@@ -828,208 +849,76 @@ def test_status_human_readable_warning(
-        assert "after setup" in err
-        assert "not present" not in err
+            setup_logging(logging.INFO, logging.DEBUG)
+        logging.log(logging.INFO, "Logging to Console")
+        out, err = capsys.readouterr()
+        assert out == ""
+        assert log_message == err.strip()
 
-    @mock.patch("uaclient.cli.util.we_are_currently_root", return_value=False)
-    @mock.patch("uaclient.log.get_user_log_file")
-    def test_console_log_configured_if_already_present(
+    def test_correct_handlers_added_to_logger(
         self,
-        m_get_user,
-        m_we_are_currently_root,
-        capsys,
         logging_sandbox,
         FakeConfig,
-        tmpdir,
+        get_logger_handler,
     ):
-        m_get_user.return_value = tmpdir.join("user.log").strpath
-        logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
-
+        console_level = logging.INFO
+        log_level = logging.DEBUG
+        logger = logging.getLogger("logger_a")
         with mock.patch(
             "uaclient.cli.config.UAConfig", return_value=FakeConfig()
         ):
-            logging.error("before setup")
-            setup_logging(logging.INFO, logging.INFO)
-            logging.error("after setup")
-
-        # 'before setup' will be in stderr, so check that setup_logging
-        # configures the format
-        _, err = capsys.readouterr()
-        assert "ERROR: before setup" not in err
-        assert "ERROR: after setup" in err
+            setup_logging(console_level, log_level, logger=logger)
+        assert len(logger.handlers) == 2
+        assert logger.handlers == get_logger_handler(
+            logger, console_level, log_level
+        )
 
-    @mock.patch("uaclient.log.get_user_log_file")
-    @mock.patch("uaclient.cli.util.we_are_currently_root", return_value=False)
-    def test_user_file_log_configured_if_not_root(
-        self,
-        m_we_are_currently_root,
-        m_log_get_user_log_file,
-        tmpdir,
-        logging_sandbox,
+    def test_previous_handlers_removed_in_logger(
+        self, logging_sandbox, FakeConfig, get_logger_handler
     ):
-        log_file = tmpdir.join("log_file")
-        m_log_get_user_log_file.return_value = log_file.strpath
+        console_level = logging.INFO
+        log_level = logging.DEBUG
+        logger = logging.getLogger("logger_b")
+        handler = logging.StreamHandler(sys.stderr)
+        handler.setLevel(logging.ERROR)
+        logger.addHandler(handler)
+        with mock.patch(
+            "uaclient.cli.config.UAConfig", return_value=FakeConfig()
+        ):
+            setup_logging(console_level, log_level, logger=logger)
 
-        setup_logging(logging.INFO, logging.INFO)
-        logging.info("after setup")
+        assert len(logger.handlers) == 2
+        assert logger.handlers == get_logger_handler(
+            logger, console_level, log_level
+        )
 
+    def test_log_file_created_if_not_present(self, logging_sandbox, tmpdir):
```

Comment:
mock the file creation methods and combine with `test_log_file_has_correct_permissions`